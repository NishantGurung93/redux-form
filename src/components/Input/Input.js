import { ErrorMessage, Field } from "formik";
import { oneOf, string, bool } from 'prop-types';

const Input = ({ label, name, placeholder, type, required }) => {
  return (
    <div className={`${name}-input`}>
      {
        required ?
          <label htmlFor={name}>{`${label}:`}<span className="asterisk"> *</span></label>
          :
          <label htmlFor={name}>{`${label}:`}</label>
      }
      <Field
        id={name}
        data-testid={`${name}-input`}
        name={name}
        placeholder={placeholder}
        type={type}
      />
      <ErrorMessage className="error" name={name} />
    </div>
  )
};

Input.propTypes = {
  type: oneOf(['text', 'email', 'password']),
  name: string,
  placeholder: string,
  label: string,
  required: bool,
};

Input.defaultProps = {
  type: 'text',
  sessioncamhide: false,
};

export default Input;
