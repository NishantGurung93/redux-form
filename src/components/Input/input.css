.input-wrapper {
  margin: 10px 0;
  position : relative;
}

.input-wrapper label {
  font-size: 1.125rem;
  display: inline-block;
  margin-bottom: 5px;
}

.input-wrapper span {
  color: red;
  position: relative;
  display: inline-block;
  margin-left: 5px;
  top: 4px;
}

.input-wrapper input {
  width: 99%;
  border: 1px solid;
}

.input-wrapper .error {
  font-size: 0.675rem;
  position: absolute;
  margin-left: 0;
  left: 0;
  top: 55px;
}
