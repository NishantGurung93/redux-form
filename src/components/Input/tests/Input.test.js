import { render } from "@testing-library/react";
import { Formik } from "formik";
import Input from "../Input";

const defaultProps = {
  label: 'Test label',
  name: 'test',
  placeholder: 'Test placeholder',
  type: 'text',
  required: false,
};

const renderComponent = props => {
  const combinedProps = {
    ...defaultProps,
    ...props,
  };

  return render(
    <Formik>
      <Input {...combinedProps} />
    </Formik>,
  );
};

describe('Input component', () => {
  it('should render the component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render * on label if required', () => {
    const { getByText } = renderComponent({ required: true });
    expect(getByText('*')).toBeInTheDocument();
  });
});
