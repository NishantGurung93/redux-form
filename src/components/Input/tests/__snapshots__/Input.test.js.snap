// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render the component 1`] = `
<div>
  <div
    class="test-input"
  >
    <label
      for="test"
    >
      Test label
    </label>
    <input
      id="test"
      name="test"
      placeholder="Test placeholder"
      type="text"
      value=""
    />
  </div>
</div>
`;
